xclim/sdba/_adjustment.py

```diff
@@ -176,3 +177,72 @@ def scaling_adjust(ds, *, group, interp, kind):
     af = u.broadcast(ds.af, ds.sim, group=group, interp=interp)
     scen = u.apply_correction(ds.sim, af, kind)
     return scen.rename("scen").to_dataset()
+
+
+def npdf_transform(ds, **kwargs):
+    """N-pdf transform : Iterative univariate adjustment in random rotated spaces.
+
+    Dataset variables:
+      ref : Reference multivariate timeseries
+      hist : simulated timeseries on the reference period
+      sim : Simulated timeseries on the projected period.
+      rot_matrices : Random rotation matrices.
+
+    Kwargs:
+      pts_dim : multivariate dimensionn name
+      base : Adjustment class
+      base_kws : Kwargs for initialising the adjustment object
+      adj_kws : Kwargs of the `adjust` call
+      n_escore : Number of elements to include in the e_score test (0 for all, < 0 to skip)
+    """
+    ref = ds.ref
+    hist = ds.hist
+    sim = ds.sim.rename(time_sim="time")
+    dim = kwargs["pts_dim"]
+
+    escores = []
+    for i, R in enumerate(ds.rot_matrices.transpose("iterations", ...)):
+
+        # Rotate to new magic space
+        # @ operator stands for matrix multiplication
+        refp = ref @ R
+        histp = hist @ R
+        simp = sim @ R
+
+        # Perform univariate adjustment in new space
+        ADJ = kwargs["base"](**kwargs["base_kws"])
+        ADJ.train(refp, histp)
+        scenhp = ADJ.adjust(histp, **kwargs["adj_kws"])
+        scensp = ADJ.adjust(simp, **kwargs["adj_kws"])
+
+        # Rotate back
+        # Confusing but works when you think about it (hint: xarray has named dims)
```

Comment:
```suggestion
        # Note that this works because the matrix multiplication is done along named 
        # dimensions. So here, the dot product is done along x', not x as in the first 
        # rotation. Because xarray broadcasts over named dimensions, this is 
        # equivalent to taking the transpose of R, the reverse rotation. 
```